### ISSUE TITLE
bug(swarm/dial) panic: Process cannot add children after being closed

### ISSUE THREAD

**Issue by briantigerchow** [jbenet/go-ipfs#754]
```
go-ipfs (feat/grandcentral-routing) λ. ipfs daemon --gcr
Initializing daemon...
10:47:26.660 CRITI supernode/: failed to bootstrap to{<peer.ID QmVy5x> []} dial attempt failed <autogenerated>:20
panic: Process cannot add children after being closed

goroutine 72 [running]:
github.com/jbenet/go-ipfs/Godeps/_workspace/src/github.com/jbenet/goprocess.(*process).AddChild(0xc2082a55e0, 0xf12398, 0xc2082a5650)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/Godeps/_workspace/src/github.com/jbenet/goprocess/impl-mutex.go:80 +0x113
github.com/jbenet/go-ipfs/Godeps/_workspace/src/github.com/jbenet/goprocess.(*process).Go(0xc2082a55e0, 0xc2083ab060, 0x0, 0x0)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/Godeps/_workspace/src/github.com/jbenet/goprocess/impl-mutex.go:91 +0x1d3
github.com/jbenet/go-ipfs/p2p/net/swarm.func·006()
        /Users/btc/go/src/github.com/jbenet/go-ipfs/p2p/net/swarm/swarm_dial.go:400 +0x23f
created by github.com/jbenet/go-ipfs/p2p/net/swarm.(*Swarm).dialAddrs
        /Users/btc/go/src/github.com/jbenet/go-ipfs/p2p/net/swarm/swarm_dial.go:402 +0x6b8

goroutine 1 [chan receive]:
github.com/jbenet/go-ipfs/Godeps/_workspace/src/github.com/jbenet/goprocess.(*process).doClose(0xc2080e7730)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/Godeps/_workspace/src/github.com/jbenet/goprocess/impl-mutex.go:151 +0x210
github.com/jbenet/go-ipfs/Godeps/_workspace/src/github.com/jbenet/goprocess.(*process).Close(0xc2080e7730, 0x0, 0x0)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/Godeps/_workspace/src/github.com/jbenet/goprocess/impl-mutex.go:117 +0xf4
github.com/jbenet/go-ipfs/core.Bootstrap(0xc208120c60, 0x4, 0x6fc23ac00, 0x2540be400, 0xc2080ab550, 0x0, 0x0, 0x0, 0x0)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/core/bootstrap.go:101 +0x307
github.com/jbenet/go-ipfs/core.(*IpfsNode).Bootstrap(0xc208120c60, 0x4, 0x6fc23ac00, 0x2540be400, 0xc2080ab550, 0x0, 0x0)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/core/core.go:344 +0x17c
github.com/jbenet/go-ipfs/core.(*IpfsNode).startOnlineServices(0xc208120c60, 0xf16d90, 0xc20811d980, 0xc2080aa0b0, 0x0, 0x0)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/core/core.go:245 +0x473
github.com/jbenet/go-ipfs/core.func·008(0xf16d90, 0xc20811d980, 0xc208120c60, 0x0, 0x0)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/core/core.go:204 +0x489
github.com/jbenet/go-ipfs/core.NewIPFSNode(0xf12e48, 0xc20809de60, 0xc208111aa0, 0x0, 0x0, 0x0)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/core/core.go:116 +0x13e
main.daemonFunc(0xf12e88, 0xc2080740c0, 0xf12fb8, 0xc20805aba0)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/cmd/ipfs/daemon.go:123 +0x6d3
github.com/jbenet/go-ipfs/commands.(*Command).Call(0xc4baa0, 0xf12e88, 0xc2080740c0, 0x0, 0x0)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/commands/command.go:98 +0x457
main.callCommand(0xf12e48, 0xc20809de60, 0xf12e88, 0xc2080740c0, 0xc4baa0, 0xc42c20, 0x0, 0x0, 0x0, 0x0)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/cmd/ipfs/main.go:325 +0xb0c
main.(*cmdInvocation).Run(0xc208063440, 0xf12e48, 0xc20809de60, 0x0, 0x0, 0x0, 0x0)
        /Users/btc/go/src/github.com/jbenet/go-ipfs/cmd/ipfs/main.go:164 +0x20e
main.main()
        /Users/btc/go/src/github.com/jbenet/go-ipfs/cmd/ipfs/main.go:135 +0x59e
```

**Comment by jbenet** [comment]
given the stacks, my first guess is that some operation is failing (maybe node bootstrap or gcr bootstrap), which in turn causes the node to close. this closes the swarm. dialAddrs is also apparently not bailing out after the swarm closes. i'll look at this now.

**Comment by jbenet** [comment]
fixed in #755
